pkg/doc/did/doc.go

```diff
@@ -777,11 +777,16 @@ type didKeyResolver struct {
 	PubKeys []PublicKey
 }
 
-func (r *didKeyResolver) Resolve(id string) (*verifier.PublicKey, error) {
+func (r *didKeyResolver) Resolve(id string) (interface{}, error) {
 	for _, key := range r.PubKeys {
 		if key.ID == id {
+			keyType, err := MapKeyTypeToKMS(key.Type)
```

Comment:
We could defer key type  mapping when it is needed (sign, verify) and not tie it to resolution. 
Did package should not know about KMS types. This should be postponed until sign or verify. You can have helper function on crypto before PubKeyBytesToHandle(pubKey.Value, pubKey.Type) is called.

Reply:
I agree with you and @troyronda . I removed this conversion. I also changed the `Type` of the `PublicKey` to be just a string as we discussed on the call with @sandrask .